relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -8,31 +9,22 @@ use relay_quotas::{DataCategory, ItemScoping, Quota, RateLimits, Scoping};
 use relay_system::Addr;
 
 use crate::envelope::SourceQuantities;
-use crate::services::outcome::{DiscardReason, Outcome, TrackOutcome};
+use crate::services::outcome::{Outcome, TrackOutcome};
 
 /// Contains all data necessary to rate limit metrics or metrics buckets.
 #[derive(Debug)]
 pub struct MetricsLimiter<Q: AsRef<Vec<Quota>> = Vec<Quota>> {
-    /// A list of aggregated metric buckets.
-    metrics: Vec<Bucket>,
+    /// A list of aggregated metric buckets with some counters.
+    buckets: Vec<SummarizedBucket>,
 
     /// The quotas set on the current project.
     quotas: Q,
 
     /// Project information.
     scoping: Scoping,
 
-    /// Binary index of metrics/buckets in the transaction namespace (used to retain).
-    transaction_buckets: Vec<bool>,
-
-    /// Binary index of metrics/buckets that encode processed profiles.
-    profile_buckets: Vec<bool>,
-
     /// The number of transactions contributing to these metrics.
-    transaction_count: usize,
-
-    /// The number of profiles contained in these metrics.
-    profile_count: usize,
+    counts: TotalEntityCounts,
```

Comment:
`counts` also counts spans now, so something like:
```diff
-/// The number of transactions contributing to these metrics.
+/// The number of performance items (transactions, spans) contributing to these metrics.
```
(can't make a suggestion on a life outside of the diff)